src/promptflow-devkit/promptflow/_cli/_pf/_service.py

```diff
@@ -155,115 +158,136 @@ def start_service(args):
+    return port
+
+
+def _validate_port(port, force_start):
+    if is_port_in_use(port):
+        if force_start:
+            message = f"Force restart the service on the port {port}."
+            if is_run_from_built_binary():
+                print(message)
+            logger.warning(message)
+            kill_exist_service(port)
+        else:
+            message = f"Service port {port} is used."
+            if is_run_from_built_binary():
+                print(message)
+            logger.warning(message)
+            raise UserErrorException(f"Service port {port} is used.")
+
+
+@contextlib.contextmanager
+def redirect_stdout_to_file(path):
+    # For msi installer, use vbs to start pfs in a hidden window. But it doesn't support redirect output in the
+    # hidden window to terminal. So we redirect output to a file. And then print the file content to terminal in
+    # pfs.bat.
+    old_stdout = sys.stdout
+    old_stderr = sys.stderr
+    try:
+        with open(path, "w") as file:
+            sys.stdout = file
+            sys.stderr = file
+            yield
+    finally:
+        sys.stdout = old_stdout
+        sys.stderr = old_stderr
+
+
+def _start_detach_service_in_win(port):
+    try:
+        import win32api
+        import win32con
+        import win32process
+    except ImportError as ex:
+        raise UserErrorException(
+            f"Please install pywin32 by 'pip install pywin32' and retry. prompt flow "
+            f"service start depends on pywin32.. {ex}"
+        )
+    command = (
+        f"waitress-serve --listen=127.0.0.1:{port} --threads={PF_SERVICE_WORKER_NUM} "
+        "promptflow._cli._pf._service:get_app"
+    )
+    logger.debug(f"Start Promptflow Service in Windows: {command}")
+    startupinfo = win32process.STARTUPINFO()
+    startupinfo.dwFlags |= win32process.STARTF_USESHOWWINDOW
+    startupinfo.wShowWindow = win32con.SW_HIDE
+    process_attributes = None
+    thread_attributes = None
+    inherit_handles = False
+    creation_flags = win32con.CREATE_NEW_PROCESS_GROUP | win32con.DETACHED_PROCESS
+    environment = None
+    current_directory = None
+    process_handle, thread_handle, process_id, thread_id = win32process.CreateProcess(
+        None,
+        command,
+        process_attributes,
+        thread_attributes,
+        inherit_handles,
+        creation_flags,
+        environment,
+        current_directory,
+        startupinfo,
+    )
+
+    win32api.CloseHandle(process_handle)
+    win32api.CloseHandle(thread_handle)
+
+
+def _start_detach_service_in_unix(port):
+    # Set host to localhost, only allow request from localhost.
+    cmd = [
+        "waitress-serve",
+        f"--listen=127.0.0.1:{port}",
+        f"--threads={PF_SERVICE_WORKER_NUM}",
+        "promptflow._cli._pf._service:get_app",
+    ]
+    logger.debug(f"Start Promptflow Service in Unix: {cmd}")
+    subprocess.Popen(cmd, stdout=subprocess.DEVNULL, start_new_session=True)
```

Comment:
how do we capture eror of this Popen failure?

Reply:
we will only capture errors when the cmd is illegal in unix
such as 1. can't find waitress command. 2. port is None, 3. get_app has bug.  

But I think we can't get detach subprocess error message.
1.
![image](https://github.com/microsoft/promptflow/assets/26239730/a10cdce6-615d-43fa-9329-dc26ea8ebbdf)
2.
![image](https://github.com/microsoft/promptflow/assets/26239730/35c0b408-c91c-48dd-927a-ddbf9cbea3b4)
3.
![image](https://github.com/microsoft/promptflow/assets/26239730/a0a5c79e-e015-48c7-9c48-5a56fb7c1f98)